mantidproject/mantid
# - Added spin state order handling to PolarizationCorrectionFredrikze Correction Algorithm

### Description of work
- Implemented InputSpinStateOrder and OutputSpinStateOrder properties.
- Updated execPA functions to dynamically handle different spin state orders.
- Extracted input and output spin state mapping logic into separate functions for better code organization and reusability.
- Ensured backward compatibility by setting default spin state orders.
- Added validation to ensure input workspace count matches the spin state entered
- Updated PolarizationEfficiencyCor algorithm to handle spin states for the PolarizationCorrectionFredrikze algorithm
- Updated Documentations

#### Purpose of work
#36981 

### To test:
- Execute the PolarizationCorrectionFredrikze Correction Algorithm
- Test with Input SpinState and/or Output Spin State, algorithm should run successfully
- Test without input SpinState and/or Output Spin State, confirm it still executes successfully
- Execute the PolarizationEfficiencyCor algorithm, running the PolarizationCorrectionFredrikze algorithm as a child. The algorithm should run successfully
